internal/common/function/function.go

```diff
@@ -161,6 +161,69 @@ func GetVeleroBackupByLabel(ctx context.Context, clientInstance client.Client, n
 	}
 }
 
+// GetActiveVeleroBackupsByLabel retrieves all VeleroBackup objects based on a specified label within a given namespace.
+// It returns a slice of VeleroBackup objects or nil if none are found.
+func GetActiveVeleroBackupsByLabel(ctx context.Context, clientInstance client.Client, namespace, labelKey, labelValue string) ([]velerov1.Backup, error) {
+	var veleroBackupList velerov1.BackupList
+	labelSelector := client.MatchingLabels{labelKey: labelValue}
+
+	if err := clientInstance.List(ctx, &veleroBackupList, client.InNamespace(namespace), labelSelector); err != nil {
+		return nil, err
+	}
+
+	// Filter out backups with a CompletionTimestamp
+	var activeBackups []velerov1.Backup
+	for _, backup := range veleroBackupList.Items {
+		if backup.Status.CompletionTimestamp == nil {
+			activeBackups = append(activeBackups, backup)
+		}
+	}
+
+	if len(activeBackups) == 0 {
+		return nil, nil
+	}
+
+	return activeBackups, nil
+}
+
+// GetBackupQueueInfo determines the queue position of the specified VeleroBackup.
+// It calculates how many queued Backups exist in the namespace that were created before this one.
+func GetBackupQueueInfo(ctx context.Context, clientInstance client.Client, namespace string, targetBackup *velerov1.Backup) (nacv1alpha1.QueueInfo, error) {
+	var queueInfo nacv1alpha1.QueueInfo
+
+	// If the target backup has no valid CreationTimestamp, it means that it's not yet reconciled by OADP/Velero.
+	// In this case, we can't determine its queue position, so we return nil.
+	if targetBackup == nil || targetBackup.CreationTimestamp.IsZero() {
```

Comment:
AFAIK creationTS field gets populated as soon as the object create request is executed by Kube API server, it does not signify whether the controller of the object reconciled on it or not.